src/SFA.DAS.RoATPService.Data/OrganisationRepository.cs

```diff
@@ -288,21 +289,24 @@ public async Task<OrganisationRegisterStatus> GetOrganisationRegisterStatus(stri
             }
         }
 
-        public async Task<IEnumerable<Engagement>> GetEngagements()
+        public async Task<IEnumerable<Engagement>> GetEngagements(GetEngagementsRequest request)
         {
             var connectionString = _configuration.SqlConnectionString;
 
             using (var connection = new SqlConnection(connectionString))
-
             {
                 if (connection.State != ConnectionState.Open)
                     await connection.OpenAsync();
 
+                var sqlIfPaginated = request.SinceEventId.Equals(0) ? string.Empty : $@" where ose.Id >= @SinceEventId
```

Comment:
Personally, I think it would be cleaner to have two endpoints and repository queries, one for 'all data' and one for 'paginated data', rather than switching behaviour based on SinceEventId = 0

Reply:
This is how other APIs work across the Apprenticeship Servuice.
We modeled this on the Submission Events API, which has the sme parameter list all of which are optional.